src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -244,6 +297,30 @@ private[table] class IndexedTableImpl(
 
   }
 
+  override def selectColumnsToIndex(
+      parameters: Map[String, String],
+      data: Option[DataFrame]): Map[String, String] = {
+    val optionalColumnsToIndex = parameters.contains(COLUMNS_TO_INDEX)
+    if (!optionalColumnsToIndex && !COLUMN_SELECTOR_ENABLED) {
+      // IF autoIndexingEnabled is disabled, and no columnsToIndex are specified we should throw an exception
+      throw AnalysisExceptionFactory.create(
+        "Auto indexing is disabled. Please specify the columns to index in a comma separated way" +
+          " as .option(columnsToIndex, ...) or enable auto indexing with spark.qbeast.index.autoIndexingEnabled=true")
+    } else if (!optionalColumnsToIndex && COLUMN_SELECTOR_ENABLED) {
+      data match {
+        case Some(df) => {
+          // If columnsToIndex is NOT present, the column selector is ENABLED and DATA is AVAILABLE
+          // We can automatically choose the columnsToIndex based on dataFrame
+          val columnsToIndex = columnSelector.selectColumnsToIndex(df)
+          parameters + (COLUMNS_TO_INDEX -> columnsToIndex.mkString(","))
+        }
+        case None =>
+          throw AnalysisExceptionFactory.create(
```

Comment:
Same issue about the behavior triggering an exception not being tested, let's either add such a test or at least create an issue to remind us that it needs to be done.

Reply:
This is related to https://github.com/Qbeast-io/qbeast-spark/pull/284#discussion_r1540833194

Reply:
Issue #295 opened